go/libraries/events/events.go

```diff
@@ -0,0 +1,78 @@
+package events
+
+import (
+	"sync"
+	"time"
+
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"github.com/google/uuid"
+
+	eventsapi "github.com/liquidata-inc/dolt/go/gen/proto/dolt/services/eventsapi_v1alpha1"
+)
+
+// EventNowFunc function is used to get the current time and can be overridden for testing.
+var EventNowFunc = time.Now
+
+func nowTimestamp() *timestamp.Timestamp {
+	now := EventNowFunc()
+	nanos := int32(now.UnixNano() % int64(time.Second))
```

Comment:
Gonna advocate for https://godoc.org/github.com/golang/protobuf/ptypes#TimestampProto, even know it can (rightly) error.